### ISSUE TITLE
[SPRING] Generating List<Object> instead of List<Foo> for [spring-cloud]

### ISSUE THREAD

**Issue by tddts** [swagger-api/swagger-codegen#9725]
##### Description
When some operation returns a list of objects, generated code is not using an appropriate model.
For example, with [this operation](https://esi.evetech.net/ui/#/Market/get_markets_region_id_orders) it generates:
```java
@ApiResponse(code = 200, message = "A list of orders", response = Object.class, responseContainer = "List")
...
ResponseEntity<List<Object>> getMarketsRegionIdOrders(....
```
when it should be (I suppose):
```java
@ApiResponse(code = 200, message = "A list of orders", response = GetMarketsRegionIdOrders200Ok.class, responseContainer = "List")
...
ResponseEntity<List<GetMarketsRegionIdOrders200Ok>> getMarketsRegionIdOrders(....
```
Everything seems to be fine for single-entity responses .
##### Swagger-codegen version

2.4.8 / latest master
(3.0.11 does not even generate models for requests/responses)

##### Swagger declaration file content or url

https://esi.evetech.net/ui/#/Market/get_markets_region_id_orders

##### Command line used for generation
```bash
java -Xmx10g -jar modules/swagger-codegen-cli/target/swagger-codegen-cli.jar generate \
  -i https://esi.evetech.net/latest/swagger.json \
  --artifact-id eve-api-client \
  --artifact-version 0.0.1-SNAPSHOT \
  -l spring \
  -o samples/eve/ \
  --library spring-cloud
```

##### Steps to reproduce

Generate API using command above.

**Comment by tddts** [comment]
So after some digging I created workaround by copying **io.swagger.models.Response** to **swagger-codegen-cli** sources and modifying **getSchema()** method, so it sets **schemaAsProperty** instead of just returning it.
Before:
```java
    @Deprecated
    public Property getSchema() {
        if (schemaAsProperty == null && schemaAsModel != null) {
            return new PropertyModelConverter().modelToProperty(schemaAsModel);
        }
        return schemaAsProperty;
    }
```
After:
```java
    @Deprecated
    public Property getSchema() {
        if (schemaAsProperty == null && schemaAsModel != null) {
            schemaAsProperty = new PropertyModelConverter().modelToProperty(schemaAsModel);
        }
        return schemaAsProperty;
    }
```
This method is actually used in **io.swagger.codegen.InlineModelResolver#flatten**, but created property does not go anywhere.

**Comment by HugoMario** [comment]
thanks for reporting this, if you are willing/available to add a PR, let me know and i can help you with reviewing and merging

**Comment by frantuma** [comment]
fixed in #9765 for master; for 3.0.0, as of latest 3.0.12-SNAPSHOT, it seems it creates responses and models ok.

Closing ticket, please reopen if you're still experiencing issues.
